File: cuckoo-hibersap-example-ejb/src/main/java/de/akquinet/jbosscc/cuckoo/example/ejb/HibersapBootstrapBean.java
package de.akquinet.jbosscc.cuckoo.example.ejb;

import org.hibersap.configuration.AnnotationConfiguration;
import org.hibersap.session.SessionManager;

import javax.annotation.PostConstruct;
import javax.annotation.PreDestroy;
import javax.ejb.Singleton;
import javax.ejb.Startup;
import javax.naming.Context;
import javax.naming.InitialContext;
import javax.naming.NamingException;
import java.util.logging.Level;
import java.util.logging.Logger;

@Singleton
@Startup
public class HibersapBootstrapBean
{
    public static final String JNDI_NAME = "java:jboss/eis/hibersap/NSP";

    private static final Logger LOG = Logger.getLogger( HibersapBootstrapBean.class.getName() );

    @PostConstruct
    public void rebindSessionManager()
    {
        SessionManager sessionManager = new AnnotationConfiguration().buildSessionManager();

        LOG.info( "Binding Hibersap SessionManager '" + sessionManager.getConfig().getName()
                + "' to JNDI name '" + JNDI_NAME + "'" );

        try
        {
            Context ctx = new InitialContext();
            ctx.rebind( JNDI_NAME, sessionManager );
        }
        catch ( NamingException e )
        {
            throw new RuntimeException( "Failed binding Hibersap SessionManager to JNDI name [" + JNDI_NAME + "]", e );
        }
    }

    @PreDestroy
    public void unbindSessionManager()
    {
        LOG.info( "Unbinding Hibersap SessionManager from JNDI name '" + JNDI_NAME + "'" );

        try
        {
            Context ctx = new InitialContext();
            ctx.unbind( JNDI_NAME );
        }
        catch ( NamingException e )
        {
            LOG.log( Level.WARNING,
                    "Failed to unbind Hibersap SessionManager binding for JNDI name [" + JNDI_NAME + "]", e );
        }
    }
}

File: src/main/kotlin/org/hibersap/customer/service/HibersapBootstrapBean.kt
package org.hibersap.customer.service

import org.hibersap.configuration.AnnotationConfiguration
import org.hibersap.ejb.util.JndiUtil
import java.util.logging.Logger
import javax.annotation.PostConstruct
import javax.annotation.PreDestroy
import javax.ejb.Singleton
import javax.ejb.Startup

const val JNDI_NAME = "java:jboss/eis/hibersap/NSP"

@Singleton
@Startup
class HibersapBootstrapBean {

    @PostConstruct
    fun rebindSessionManager() {
        val sessionManager = AnnotationConfiguration("NSP").buildSessionManager()
        JndiUtil.rebindSessionManager(sessionManager, JNDI_NAME);
    }

    @PreDestroy
    fun unbindSessionManager() {
        JndiUtil.unbindSessionManager(JNDI_NAME)
    }

    companion object {
        private val LOG = Logger.getLogger(HibersapBootstrapBean::class.java.name)
    }
}